### PR TITLE
Add redux

### PR DESCRIPTION
**Pull request by Berkay-akbas**
## Changes 
- Created directories for redux.
- Actions and reducer for adding and removing implemented.
- Action and reducer for categories component implemented.
- Updated README.md

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const ADD = 'ADD_BOOK';
+const REMOVE = 'REMOVE_BOOK';
```
Comment (zhadier) [inline src/redux/books/books.js:2]: - [ ] Inorder to have your actions follow the DUCK pattern, you need to ensure that you prepend the action with the file path. For example
``` js
const ADD_BOOK = bookstore/books/ADD_BOOK
```
Kindly rename all of your actions to conform to the ducks pattern

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+// function nextBookId(books) {
+//   const maxId = books.reduce((maxId, book) => Math.max(book.id, maxId), -1)
+//   return maxId + 1
+// }
```
Comment (zhadier) [inline src/redux/books/books.js:7]: - [ ] Kindly remove commented out code from your files as it is not optimal for maintainability and manageability.

**Review by zhadier — CHANGES_REQUESTED** [review]
# Hi @Berkay-akbas  👋 

Good job so far! 🌤️ 
There are some issues that you still need to work on to go to the next project but you are almost there! 🍓 



## Highlights
✔️ No linter errors
✔️ Github flow was set up correctly
✔️ Professional pull request title and description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question @zhadier so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Berkay-akbas,

✔️ Use of ducks patterns

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**